syscalls.c

```diff
@@ -360,10 +360,15 @@ int syscalls_nsleep(void *ustack)
 
 	if (ret == -EINTR) {
 		proc_gettime(&stop, NULL);
-		diff = stop - start;
-		if (diff < us) {
-			*sec = diff / (1000 * 1000);
-			*nsec = (diff % (1000 * 1000)) * 1000;
+		elapsed = stop - start;
+		unslept = us - elapsed;
+		if ((long long)unslept <= 0LL) {
```

Comment:
This check is UB - cast might cause integer overflow, compiler has right to optimize this condition to false (as no integer overflow may occur per standard)

Reply:
I changed it and also uncertained why time_t is unsigned. If it was signed all would have been much easier.